### PR TITLE
API Add publish / unpublish actions to asset admin

### LINKED ISSUE #169: Implement file publish control actions
## Acceptance criteria

 * Action toolbar and options available on the file details match closely to the page button (see Paul's designs for more details)

## Tasks

 * [X] Implement unpublish action on "File" section (no indication in UI at the moment)
 * [X] Remove unpublish action when file is already unpublished
 * [ ] Merge https://github.com/silverstripe/silverstripe-asset-admin/pull/280

## Related

 * https://github.com/silverstripe/silverstripe-framework/issues/5529#issuecomment-218970412
 * https://github.com/silverstripe/silverstripe-framework/issues/5611
 * [Design UX Patterns](https://github.com/silverstripe/silverstripe-asset-admin/issues/219)

### PR DESCRIPTION
**Pull request by tractorcow**
Fixes #169

Requires https://github.com/silverstripe/silverstripe-framework/pull/6119 to be merged first

### CODE REVIEW

File: code/Controller/AssetAdmin.php [hunk 1]
```diff
@@ -873,4 +907,72 @@ protected function getUpload()
...
+            // Create save button
+            $saveAction = FormAction::create('save', _t('CMSMain.SAVE', 'Save'));
+            $saveAction->setIcon('save');
+            $actions->push($saveAction);
+
+            // Folders are automatically published
+            if ($file->canPublish() && (!$file instanceof Folder)) {
+                $publishText = _t('SilverStripe\\AssetAdmin\\Controller\\AssetAdmin.PUBLISH_BUTTON', 'Publish');
+                $publishAction = FormAction::create('publish', $publishText);
+                $publishAction->setIcon('save');
```
Comment (flamerohr) [inline code/Controller/AssetAdmin.php]: icon for publish should be a rocket, or at least different to save:
https://projects.invisionapp.com/boards/9GX68PTXUQFH/#/1547737/71606997

File: code/Controller/AssetAdmin.php [hunk 2]
```diff
@@ -873,4 +907,72 @@ protected function getUpload()
...
+                $addToCampaignAction
+            ];
+            // Add unpublish if available
+            if ($file->isPublished() && $file->canUnpublish()) {
+                $unpublishText = _t(
+                    'SilverStripe\\AssetAdmin\\Controller\\AssetAdmin.UNPUBLISH_BUTTON',
+                    'Unpublish'
+                );
+                $unpublishAction = FormAction::create('unpublish', $unpublishText);
+                $unpublishAction->setIcon('trash-bin');
```
Comment (flamerohr) [inline code/Controller/AssetAdmin.php]: Not sure about using trash-bin for unpublish as well... I'd suggest something different from delete like `cancel-circled`

This one is up for debate @clarkepaul might have a better idea

File: code/Controller/AssetAdmin.php [hunk 3]
```diff
@@ -873,4 +907,72 @@ protected function getUpload()
...
+
+            // Build popover menu
+            $popoverField = PopoverField::create($popoverActions);
+            $popoverField->setPlacement('top');
+            $actions->push($popoverField);
+        } else {
+            $actions->push($deleteAction);
+        }
+
+        $this->extend('updateCMSActions', $actions);
```
Comment (flamerohr) [inline code/Controller/AssetAdmin.php]: Rename extension hook to `updateFileEditActions` as they could potentially be more "Action" methods later on.

**Comment by tractorcow** [comment]
Fixed @flamerohr

**Comment by flamerohr** [comment]
Looks good, I've approved the changes made, so ready for merge

**Pull request merged**